mondey_backend/src/mondey_backend/routers/users.py

```diff
@@ -152,4 +156,59 @@ def update_milestone_answer(
             session.commit()
         return milestone_answer
 
+    # Endpoints for answers to user question
+    @router.get("/user-answers/", response_model=list[UserAnswerPublic])
+    def get_current_user_answers(
+        session: SessionDep, current_active_user: CurrentActiveUserDep
+    ):
+        answers = session.exec(
+            select(UserAnswer).where(col(UserAnswer.user_id) == current_active_user.id)
+        ).all()
+
+        user_answers = [
+            UserAnswerPublic(
+                answer=answer.answer,
+                non_standard=answer.non_standard,
+                question_id=answer.question_id,
+            )
+            for answer in answers
+        ]
+        return user_answers
+
+    @router.put("/user-answers/", response_model=list[UserAnswerPublic])
+    def update_current_user_answers(
+        session: SessionDep,
+        current_active_user: CurrentActiveUserDep,
+        new_answers: list[UserAnswerPublic],
+    ):
+        if session is None:
+            raise HTTPException(401, detail="no session")
+
+        stored_answers = {
+            answer.question_id: answer
+            for answer in session.exec(
+                select(UserAnswer).where(UserAnswer.user_id == current_active_user.id)
+            ).all()
+        }
+
+        if len(stored_answers) > 0:
+            for new_answer in new_answers:
+                stored_answer = stored_answers[new_answer.question_id]
+                stored_answer.answer = new_answer.answer
+                stored_answer.non_standard = new_answer.non_standard
+                session.add(stored_answer)
+        else:
+            for new_answer in new_answers:
+                session.add(
+                    UserAnswer(
+                        user_id=current_active_user.id,
+                        question_id=new_answer.question_id,
+                        answer=new_answer.answer,
+                        non_standard=new_answer.non_standard,
+                    )
+                )
```

Comment:
this logic looks a bit fragile - e.g. what happens if a new question is added by the admins?
Assuming we add the id to UserAnswerPublic then we could just iterate the provided answers and for each either get and update the existing one in the database, or if it doesn't exist create one, something like (untested):
```python
    for new_answer in new_answers:
        db_new_answer = session.get(UserAnswer, new_answer.id)
        if not db_new_answer:
            db_new_answer = UserAnswer.model_validate(new_answer, update={"user_id": current_active_user.id})
            session.add(db_new_answer)
        else:
            for key, value in new_answer.model_dump().items():
                setattr(db_new_answer, key, value)
```

Reply:
changed that into a tuple key as we discussed yesterday.